internal/client/job.go

```diff
@@ -55,19 +45,78 @@ func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI) (*pb.J
 	return c.doJobMonitored(ctx, job, ui, nil)
 }
 
+// setupLocalJobSystem does the pre-work required to run jobs locally. This includes:
```

Comment:
Extra scrutiny here please. This whole function smells, but we're going to need to execute this bundle of logic _somewhere_ - either in the status CLI command (if we don't move the runner startup logic), or in the logs and exec operations (as this pr proposes). This is the least-worst option I've come up with, but i'm certainly open to better ideas.